tutorials/CommandsArguments.md

```diff
@@ -0,0 +1,64 @@
+
+- The first one (`<>`) defines **required** arguments, the command will not run until these arguments are processed correctly, they can be anywhere.
+- The second one (`[]`) defines **optional** arguments, they're ignored when the input is incorrect and said message argument may drift to the next. That is, if you have an optional argument taking `1` and the next argument is `2` (both literals) and you pass `2`, the first parameter will be ignored while the second will take it (do not confuse with arguments).
+- The latter (`()`) defines **semi-required** arguments, they are an hybrid between both and they can turn into required arguments when a value is given.
+
+> Semi-required arguments are implemented in Klasa since 0.5.0, check PR [here](https://github.com/dirigeants/klasa/pull/162).
+
+But in difference to the CLI command format standard, we have **our own standard**, and we implement more things such the aforementioned **semi-required** arguments, and more.
+
+A more in-depth vision of arguments is that they are *arrays* of "possibles", that is, even if we aren't using the `|` token which defines extra types/possibles, there's always one for each argument. So `<nameOne:typeOne|nameTwo:typeTwo>` is an argument with **two possibles**. Each possible has their own structure and options:
+
+- They are grouped into a "Tag" or "argument", a Tag is a parsed argument that contains these multiple possibles, therefore, possibles inherit the required level from them. For example, if a Tag is required and has 3 possibles, the __internal parser__ will try to parse all three until it finds one that resolves, in the opposite case (no match), it'll ask for one of them.
+- They have their own names, types, and options. You can have a Tag with multiple types, for example, `<target:member|userID:string>` is a Tag with two possibles, one that will try to resolve into a `GuildMember` instance, or a `string` if it did not match (be careful as the `string` type is "lazy": it always resolve, so they should put last).
+
+## Repeating arguments
+
+Unlike in CLI, repeating arguments are **only** written as `[...]` and goes at the end of usage, making the last argument repeating. What does this mean? Let's say you want to have a [**choice**](https://github.com/dirigeants/klasa-pieces/blob/master/commands/Fun/choice.js) command, then you may want the user to write in multiple arguments, more likely with `,` as **usageDelim** to separate them. They you may put `<choice:string> [...]`, which makes the command take one or more parameters of this type.
+
+Unlike normal arguments, a repeating one **may not** return a single value but multiple, all of the same type (or of the first one that resolves if using multiple possibles), that being said, it's advisable to use the [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator) for arrays if you have multiple arguments.
```

Comment:
This is confusing, as if the previous argument is required, it will always return at least one value; conversely, if the previous argument is optional, it may return no values at all.